src/cmd.c

```diff
@@ -1089,6 +1090,139 @@ static void pctx(atci_param_t *param)
+    i++;
+    gpio_write(RADIO_DIO_2_PORT, RADIO_DIO_2_PIN, (i % 2) == 0);
+
+    /*
+        static bool pin = true;
+
+        // toggle every second when datarate is 4800
+        if(i % 4800 == 0)
+        {
+            atci_printf("X");
+            pin = !pin;
+        }
+
+        gpio_write(RADIO_DIO_2_PORT, RADIO_DIO_2_PIN, pin);
+
+    */
+}
+
+
+static void cm(atci_param_t *param)
+{
+    (void)param;
+
+    // Example preamble 5, datarate 4800
+    // AT$CM 868300000,250000,4800,5,-10,3
+    uint32_t freq, timeout, fdev, datarate, preamble;
+    int32_t power;
+
+    if (!atci_param_get_uint(param, &freq)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &fdev)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &datarate)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &preamble)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_int(param, &power)) abort(ERR_PARAM);
+    if (!atci_param_is_comma(param)) abort(ERR_PARAM);
+    if (!atci_param_get_uint(param, &timeout)) abort(ERR_PARAM);
+
+    log_debug("$CM: freq=%ld fdev=%ld datarate=%ld preamble=%ld power=%ld timeout=%ld\r\n", freq, fdev, datarate, preamble, power, timeout);
+
+    // Call this to set MacState to LORAMAC_TX_RUNNING to keep TX running infinitely
+    MlmeReq_t mlr = { .Type = MLME_TXCW };
+    mlr.Req.TxCw.Timeout = timeout;
+    mlr.Req.TxCw.Frequency = freq;
+    mlr.Req.TxCw.Power = power;
+    abort_on_error(LoRaMacMlmeRequest(&mlr));
+
+    timeout = ( uint32_t )timeout * 1000;
+
+    SX1276SetOpMode( RF_OPMODE_STANDBY );
+
+    SX1276SetChannel(freq);
+    SX1276SetTxConfig( MODEM_FSK, power, fdev, 0, datarate, 0, preamble, false, false, 0, 0, 0, timeout );
+
+    // Disable packet mode
+    SX1276Write( REG_PACKETCONFIG2, ( SX1276Read( REG_PACKETCONFIG2 ) & RF_PACKETCONFIG2_DATAMODE_MASK ) );
+    // Disable radio interrupts, set clock on DIO1
+    SX1276Write( REG_DIOMAPPING1, RF_DIOMAPPING1_DIO0_11 | RF_DIOMAPPING1_DIO1_00 );
+    SX1276Write( REG_DIOMAPPING2, RF_DIOMAPPING2_DIO4_10 | RF_DIOMAPPING2_DIO5_10 );
+
+    // Reconfigure CLK falling interrupt. SX samples on rising edge
+    GPIO_InitTypeDef cfg_dio1 = {
+        .Mode = GPIO_MODE_IT_FALLING,
+        .Pull = GPIO_PULLUP,
+        .Speed = GPIO_SPEED_HIGH
+    };
+    gpio_init(RADIO_DIO_1_PORT, RADIO_DIO_1_PIN, &cfg_dio1);
+
+    // Configure data pin DIO2 as an output
+    GPIO_InitTypeDef cfg_dio2 = {
+        .Mode = GPIO_MODE_OUTPUT_PP,
+        .Pull = GPIO_NOPULL,
+        .Speed = GPIO_SPEED_HIGH
+    };
+    gpio_init(RADIO_DIO_2_PORT, RADIO_DIO_2_PIN, &cfg_dio2);
+
+    // Rewire interrupts
+    DioIrqHandler *DioIrq[] = { NULL, cm_clk_irq_handler, NULL, NULL, NULL, NULL };
+    SX1276IoIrqInit(DioIrq);
+
+    SX1276SetOpMode( RF_OPMODE_TRANSMITTER );
```

Comment:
Would it make sense to use `SX1276SetTx` here instead? That appears to be a higher-layer function which also takes care of timer initialization. The function is declared as static, but I could introduce a local modification to LoRaMac-node to export it. I do that for other stuff anyway, so keeping one more local modification isn't a big deal.

Reply:
Yes, I will just remove the static keyword in my fork of LoRaMac-node over at https://github.com/janakj/LoRaMac-node. I maintain branches with custom fixes and improvements there. Those are then merged into the master branch which is then used as a submodule in https://github.com/hardwario/lora-modem-abz. The process is a bit tedious, but it allows me to keep custom  modifications on top of the most recent LoRaMac-node upstream.

Reply:
I have made the modification to https://github.com/janakj/LoRaMac-node. Cd into lib/loramac-node, try to do git pull and then checkout the master branch to update the submodule. This should (hopefully) give you access to `SX1276SetTx`.

Reply:
Now I understand. It seems you missed to remove `static` in the function itself.
![image](https://user-images.githubusercontent.com/497212/179930547-bff7edbe-41cd-4126-b2c6-07e102ccb358.png)
Making PRs directly in github is great! https://github.com/janakj/LoRaMac-node/pull/1

Reply:
I did remove the static keyword: https://github.com/janakj/LoRaMac-node/commit/7ec078cd619b9e8a222bf622f3bd0fed150b3e89

Or did you mean something else?